File: readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

int count_letters(string text);
int count_words(string text);

int main(void)
{
    //Prompt user for input text.
    string text = get_string("Text: ");

    int letters = count_letters(text);
    printf("%i letter(s)\n", letters);

    int words = count_words(text);
    printf("%i word(s)\n", words);

    return 0;
}

int count_letters(string text)
// Returns nbr of letters in string <text>.
{
    int cnt = 0;
    for (int i=0, n = strlen(text); i < n ; i++)
    {
        if (isalpha(text[i]))
        {
            cnt++;
        }
    }
    return cnt;
}

int count_words(string text)
// Return nbr of words in <text>.
{
    int cnt = 0;
    for (int i=0, n = strlen(text); i < n ; i++)
    {
        if (isblank(text[i]))
        {
            cnt++;
        }
    }
    cnt++;
    return cnt;
}

File: pset6/credit/readability/readability.py
"""readability.py
Computes the approximate grade level needed to comprehend some text."""
import re


def main():
    # Prompt user for input text
    text = input("Text: ")

    # Count letters, words, sentences in text
    l = count_letters(text)
    w = count_words(text)
    s = count_sentences(text)

    # Compute mean n of letters per 100 words
    L = 100 * l / w

    # Compute mean n of sentences per 100 words
    S = 100 * s / w

    # Compute Coleman-Liau index
    idx = round(0.0588 * L - 0.296 * S - 15.8)

    # Print grade readability
    if idx < 1:
        print("Before Grade 1")
    elif idx > 16:
        print("Grade 16+")
    else:
        print(f"Grade {idx}")
    

def count_letters(text: str) -> int:
    """Return n of letters in <text>"""
    return len(re.findall("([a-zA-Z])", text))


def count_words(text: str) -> int:
    """Return n of words in <text>"""
    return len(text.split())


def count_sentences(text: str) -> int:
    """Return n of sentences in <text>"""
    return len(re.findall("([.|!|?])", text))


if __name__ == "__main__":
    main()